toolset/setup/linux/setup_util.py

```diff
@@ -1,8 +1,62 @@
 import re
 import os
+import sys
 import subprocess
 import platform
 
+from threading import Thread
+from Queue import Queue, Empty
+
+class NonBlockingStreamReader:
```

Comment:
By the way, I emailed the original author o fthis class and he confirmed we are free to use it as part of this project (I had to double-check because there's no license statement on it)